### PR TITLE
Networks information in inspectContainer

### PR DESCRIPTION
**Pull request by damianopezzotti**
Returns information about networks associated to the container in _inspectContainer_ flow, as explained [here](https://docs.docker.com/engine/reference/api/docker_remote_api_v1.21/#inspect-a-container)

### CODE REVIEW

**Comment by davidxia** [comment]
@damianopezzotti Thanks for the PR. I'll take a look soon. Could you squash into one commit?

**Comment by damianopezzotti** [comment]
@davidxia squash done

File: src/main/java/com/spotify/docker/client/messages/NetworkSettings.java [hunk 1]
```diff
@@ -215,6 +228,17 @@ public Builder macAddress(final String macAddress) {
       return this;
     }
 
+    public Builder networks(final Map<String, AttachedNetwork> networks) {
+      if (networks != null) {
+        final ImmutableMap.Builder<String, AttachedNetwork> builder = ImmutableMap.builder();
```
Comment (davidxia) [inline src/main/java/com/spotify/docker/client/messages/NetworkSettings.java]: I think you can replace everything in this `if` with `this.networks = ImmutableMap.copyOf(networks);`

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java [hunk 2]
```diff
@@ -2108,6 +2109,17 @@ public void testNetworksConnectContainer() throws Exception {
...
+    final ContainerInfo containerInfo = sut.inspectContainer(containerCreation.id());
+    assertThat(containerInfo.networkSettings().networks().size(), is(2));
+    final AttachedNetwork attachedNetwork =
+        containerInfo.networkSettings().networks().get(networkName);
+    assertThat(attachedNetwork, is(notNullValue()));
+    assertThat(attachedNetwork.endpointId(), is(notNullValue()));
+    assertThat(attachedNetwork.gateway(), is(notNullValue()));
+    assertThat(attachedNetwork.ipAddress(), is(notNullValue()));
+    assertThat(attachedNetwork.ipPrefixLen(), is(notNullValue()));
+    assertThat(attachedNetwork.macAddress(), is(notNullValue()));
```
Comment (davidxia) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: Can we also test `ipv6Gateway()`, `globalIPv6Address()`, and `globalIPv6PrefixLen()` here?

**Comment by davidxia** [comment]
It'd be nice to add tests for attaching the container to a network and inspecting the container to see it works, but we can add this later.

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java (hunk 2 above)
Comment (damianopezzotti) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: I think that result depends from machine setup (if ipv6 is enabled or not). On my machine `ipv6Gateway()` and `globalIPv6Address()` return void String and `globalIPv6PrefixLen()` returns 0.
I added a check that verifies that `ipv6Gateway()` and `globalIPv6Address()` return not null values and `globalIPv6PrefixLen()` >=0. Is it ok?

**Comment by damianopezzotti** [comment]
@davidxia Test `testNetworksConnectContainer` creates a network, attach a container to the network and then inspect it.

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java (hunk 2 above)
Comment (davidxia) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: Yea that seems good.

Comment (damianopezzotti) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: pushed

File: src/main/java/com/spotify/docker/client/messages/NetworkSettings.java (hunk 1 above)
Comment (damianopezzotti) [inline src/main/java/com/spotify/docker/client/messages/NetworkSettings.java]: pushed

**Comment by davidxia** [comment]
@damianopezzotti Thanks!

**Pull request merged**